code/datums/storage/storage.dm

```diff
@@ -401,14 +401,19 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 	if(parent.loc != user && parent.loc.loc != user) //loc.loc handles edge case of storage attached to an item attached to another item (modules/boots)
 		return COMPONENT_NO_MOUSEDROP
 
-	if(!user.restrained() && !user.stat)
-		switch(over_object.name)
-			if("r_hand")
-				INVOKE_ASYNC(src, PROC_REF(put_item_in_r_hand), source, user)
-				return COMPONENT_NO_MOUSEDROP
-			if("l_hand")
-				INVOKE_ASYNC(src, PROC_REF(put_item_in_l_hand), source, user)
-				return COMPONENT_NO_MOUSEDROP
+	if(user.restrained() || user.stat)
+		return COMPONENT_NO_MOUSEDROP
+
+	return put_storage_in_hand(source, over_object, user)
```

Comment:
you're returning the proc instead of COMPONENT_NO_MOUSEDROP, which is an issue since you're making the proc do a do_after.